docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,121 @@
+  * The **Control Application** is a program, usually written by an **Automation Engineer** in a domain specific graphical programming language (e.g. "Ladder Diagram" or "Function block diagram"). 
+    * It is developed with a specific Development Software on a PC (e.g. __STEP 7__ or __CODESYS__), and downloaded to the PLC.
+  * The **RTS** (**R**un **T**ime **S**ystem) is the basic software component on a PLC, that
+    * receives and accepts the **Control Application** from the Development Software on a PC
+    * manages and observes the cyclic execution of the **Control Application**
+  * **Sensors** and **Actuators** enable the PLC to observe and control physical behaviours on the shopfloor.
+    * can built-in in the PLC device, 
+      or connected to the PLC via some **Fieldbus** (e.g. Modbus, CANopen, PROFINET, EtherCAT, ...)
+    * can be simple inputs/outputs (as digital signals from a light barrier, or analouge values like temperature or pressure);
+      or complex equipments as robotic arms or other PLCs
+  * The **Process Image** is a block in the memory, that reflects inputs and outputs of all connected **Sensors** and **Actuators**
+    * it contains an _input area_ and an _output area_, where both areas are arrays of **data points**
+    * each **data point** carries a value of an input or output
+    * it is managed by the RTS during the cyclic execution of the **Control Application**, as steps below:
+      * Step 1: RTS reads all inputs from Sensor's and Actuator's, and stores all read input values to the **Process Image's** _input area_ 
+      * Step 2: RTS executes once the control application, that operates on the **Process Image's** _input_ and _output area_
+      * Step 3: RTS reads all output values from the **Process Image's** _output area_, and writes all read values to Sensor's and Actuator's
+      * then, next cycle starts with Step 1
+
+* **SoC**-based (**S**ystem **o**n **C**hip) Embedded System:
+  * TODO
+
+* **Microcontroller**-based Embedded System:
+  * TODO
+
+## thin-edge device concept
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+All devices are categorized as below:
+  * the **main-device** is the device thin-edge is running on
+  * all devices connected to the **main-device** are referred as **external child-devices**
+  * each device has a unique **device-id**
+
+## thin-edge data concepts
+
+**thin-edge** provides different devices in a standardized representation to the cloud. Therefore **thin-edge** provides different kinds of data concepts.
+
+  * **Measurements**:
+    * contain numeric data produced by sensors (like temperature readings) or calculated data based on information from the **control application**.
+    * a **measurements** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Variables**:
```

Comment:
The name `variables` sounds a bit vague. Why not use `set points` as that's a more familiar term in the domain?

Reply:
> I would use the term command as this is something sent to the device for action.

I applied that suggestion.